### PR TITLE
closes #3

### LINKED ISSUE #3: Feat: clone repository, complile, run tests, save result to database.

### PR DESCRIPTION
**Pull request by henrikglass**
Closes #3

### CODE REVIEW

**Comment by henrikglass** [comment]
This works for public repositories. 

Maybe we should write a ci.sh for this repository, for demonstration purposes?

**Comment by btellstrom** [comment]
That is a lot of info in the `.json` file. Should we add something to `History` to further identify the repo perhaps?

Looks good to me otherwise.

And yes, we should write a `ci.sh` for this repo, it is stated in the grading criteria that *we should eat our own dogfood*.

File: src/test/test_repo_tester.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    Should, given a pull request type event, clone the repo,                    
+    switch to the correct branch, run "ci.sh" in the root                       
+    of the repository, remove the repo, and finally return                      
+    the exit code of "ci.sh". for json_pull_request_example.txt
+    the exit code should be 10.
+    """                                                                         
+    def test_1(self):                                                           
+        with open('test/json_pull_request_example.txt', 'r') as payloadfile:         
+            payload = payloadfile.read()                                        
+            self.assertTrue(repo_test(payload) == 10)
```
Comment (ffsinger) [inline src/test/test_repo_tester.py:16]: Why should the exit code be 10? In the doc it says that it is either 0, 1 or 2

Comment (henrikglass) [reply src/test/test_repo_tester.py:16]: The comment says the exit code should be 0, 1 or 2, but in reality the function will return whatever exit code ci.sh returns. In my demo repo the contents of ci.sh is:

```
echo "hello"
exit 10
```

A proper ci.sh should return 0, 1 or 2.

Comment (btellstrom) [reply src/test/test_repo_tester.py:16]: Can we ensure that the `ci.sh` can do that easily by the tests, or is it to be done manually?

Comment (henrikglass) [reply src/test/test_repo_tester.py:16]: I'm 95% sure that python unittests return exit status 1 if a test fail, and 0 if all tests succeed.
I'm 99% sure that python static syntax checking returns 1 if it fails and 0 if it succeeds. The ci.sh should be very short and simple. It should look something like this:

```
# static syntax check
python -m py_compile src/main/*.py
if [ $? -eq 1 ]
then
    exit 2
fi

# run tests
python -m unittest discover test/
if [ $? -eq 1 ]
then
    exit 1
fi

# everything succeeds
exit 0
```

I'm 5% sure the syntax above is correct

**Comment by henrikglass** [comment]
Could someone merge this? Or should I change something?

**Comment by btellstrom** [comment]
Sorry, didn't see your response. In that case, it's fine. I'll merge it.

**Pull request merged**